pkg/webhook/admission/pod/storage_initializer_injector.go

```diff
@@ -102,213 +178,381 @@ func (mi *StorageInitializerInjector) InjectModelcar(pod *corev1.Pod) error {
+			}
 
-		initContainer.Env = append(initContainer.Env, corev1.EnvVar{
-			Name:  constants.CaBundleVolumeMountPathEnvVarKey,
-			Value: caBundleVolumeMountPath,
-		})
+			caBundleVolumeMountPath := params.Config.CaBundleVolumeMountPath
+			if caBundleVolumeMountPath == "" {
+				caBundleVolumeMountPath = constants.DefaultCaBundleVolumeMountPath
+			}
+
+			for _, envVar := range initContainer.Env {
+				if envVar.Name == s3.AWSCABundleConfigMap {
+					caBundleConfigMapName = envVar.Value
+				}
+				if envVar.Name == s3.AWSCABundle {
+					caBundleVolumeMountPath = filepath.Dir(envVar.Value)
+				}
+			}
 
-		caBundleVolume := corev1.Volume{
-			Name: CaBundleVolumeName,
-			VolumeSource: corev1.VolumeSource{
-				ConfigMap: &corev1.ConfigMapVolumeSource{
-					LocalObjectReference: corev1.LocalObjectReference{
-						Name: caBundleConfigMapName,
+			initContainer.Env = append(initContainer.Env, corev1.EnvVar{
+				Name:  constants.CaBundleConfigMapNameEnvVarKey,
+				Value: caBundleConfigMapName,
+			})
+
+			initContainer.Env = append(initContainer.Env, corev1.EnvVar{
+				Name:  constants.CaBundleVolumeMountPathEnvVarKey,
+				Value: caBundleVolumeMountPath,
+			})
+
+			caBundleVolume := corev1.Volume{
+				Name: CaBundleVolumeName,
+				VolumeSource: corev1.VolumeSource{
+					ConfigMap: &corev1.ConfigMapVolumeSource{
+						LocalObjectReference: corev1.LocalObjectReference{
+							Name: caBundleConfigMapName,
+						},
 					},
 				},
-			},
+			}
+
+			caBundleVolumeMount := corev1.VolumeMount{
+				Name:      CaBundleVolumeName,
+				MountPath: caBundleVolumeMountPath,
+				ReadOnly:  true,
+			}
+
+			params.PodSpec.Volumes = append(params.PodSpec.Volumes, caBundleVolume)
+			initContainer.VolumeMounts = append(initContainer.VolumeMounts, caBundleVolumeMount)
 		}
 
-		caBundleVolumeMount := corev1.VolumeMount{
-			Name:      CaBundleVolumeName,
-			MountPath: caBundleVolumeMountPath,
-			ReadOnly:  true,
+		// Update initContainer (container spec) from a storage container CR if there is a match,
+		// otherwise initContainer is not updated.
+		// Priority: CR > configMap
+		if params.StorageContainerSpec != nil {
+			err := mergeContainerSpecs(initContainer, &params.StorageContainerSpec.Container)
+			if err != nil {
+				return err
+			}
 		}
+	}
+
+	return nil
+}
 
-		pod.Spec.Volumes = append(pod.Spec.Volumes, caBundleVolume)
-		initContainer.VolumeMounts = append(initContainer.VolumeMounts, caBundleVolumeMount)
+// InjectStorageInitializer injects an init container to provision model data
+// for the serving container in a unified way across storage tech by injecting
+// a provisioning INIT container. This is a workaround because Knative does not
+// support INIT containers: https://github.com/knative/serving/issues/4307
+func (mi *StorageInitializerInjector) InjectStorageInitializer(ctx context.Context, pod *corev1.Pod) error {
+	// Only inject if the required annotations are set
```

Comment:
nit: add a comment that this is only for the old single storage uri case and this is moved to the controller for `storageUris`